tests/evm.rs

```diff
@@ -166,3 +167,134 @@ contract testing  {
 
     assert!(!ns.diagnostics.any_errors());
 }
+
+#[test]
+fn ethereum_solidity_tests() {
+    let error_matcher = regex::Regex::new(r"// ----\r?\n// \w+Error( \d+)?:").unwrap();
+
+    let semantic_tests = WalkDir::new(
+        Path::new(env!("CARGO_MANIFEST_DIR"))
+            .join("testdata/solidity/test/libsolidity/semanticTests"),
+    )
+    .into_iter()
+    .collect::<Result<Vec<_>, _>>()
+    .unwrap();
+
+    let syntax_tests = WalkDir::new(
+        Path::new(env!("CARGO_MANIFEST_DIR"))
+            .join("testdata/solidity/test/libsolidity/syntaxTests"),
+    )
+    .into_iter()
+    .collect::<Result<Vec<_>, _>>()
+    .unwrap();
+
+    let errors: usize = semantic_tests
+        .into_iter()
+        .chain(syntax_tests.into_iter())
```

Comment:
Could be worthwhile a parallel iter with rayon?

Reply:
Nice.

Has been a while since I last used rayon, but sometimes you need to use [the parallel bridge](https://docs.rs/rayon/latest/rayon/iter/trait.ParallelBridge.html), might be such a case. But whatever, done is better than perfect :+1:

Reply:
Right, I did not know about that - it works! I'll add this in another PR